fix(curriculum): Hide Ribbons for Legacy Curriculum Sections

VP-66 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #54935

<!-- Feel free to add any additional description of changes below this line -->

This pull request introduces a UI adjustment to the curriculum map component. It specifically omits the completion ribbons for the legacy courses from the super blocks. 

Files Modified
index.tsx

Changes
The main change involves adding a check in the MapLi component to determine if a super block belongs to the legacy curriculum. If so, the completion ribbon is not rendered for that super block.

Here are the changes in action:
<img width="897" alt="Screenshot 2024-05-23 at 6 37 37 pm" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/99856161/09298d34-152a-4cf8-afe8-4dd323edc404">

VP-66 committed:
Removed Ribbon for the legacy courses

lasjorg commented:
I would suggest using a prop, just like we do for `showProgressionLines`

So something like `showRibbonIcon`, set it to true by default and pass it `false` in the map for the `SuperBlockStages.Legacy`

VP-66 committed:
Modified the design to address the feedback

VP-66 committed:
Fixed mistake of adding console.log for debugging

VP-66 commented:
Hey @lasjorg , thank you so much for your feedback and I can definetely see how we can design this better. I have addressed your feedback by now creating a boolean flag called `showRibbonIcon` which is set to true by default, however, in the map function for `SuperBlockStages.Legacy` it will be set to false. 

I have tested this locally and the results are as we expect. 

Thank you once again!

ilenia-magoni commented:
Closing as stale, if you want to continue working on this leave a comment and it can be reopened.
